frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -421,13 +423,17 @@ const IR::ParserState* ProgramStructure::convertParser(const IR::V1Parser* parse
     const IR::Expression* select = nullptr;
     if (parser->select != nullptr) {
         auto list = new IR::ListExpression(parser->select->srcInfo, {});
-        std::vector<int> sizes;
+        std::vector<std::pair<bool, int>> signAndSizes;
```

Comment:
How about just using a `std::vector<const IR::Type::Bits *>` -- when the type is a `Bits`, put it directly in the vector, otherwise just get the bits type.